### PR TITLE
feat: Add support for autoscaling

### PR DESCRIPTION
**Pull request by Mariatta**
- Add the parameters min_serve_nodes, max_serve_nodes, and cpu_utilization_percent
- Create disable_autoscaling function
- Update documentation and tests

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigtable/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: google/cloud/bigtable/cluster.py [hunk 1]
```diff
@@ -85,6 +89,28 @@ class Cluster(object):
                    :data:`google.cloud.bigtable.enums.Cluster.State.CREATING`.
                    :data:`google.cloud.bigtable.enums.Cluster.State.RESIZING`.
                    :data:`google.cloud.bigtable.enums.Cluster.State.DISABLED`.
+
+    :type min_serve_nodes: int
+    :param min_serve_nodes: (Optional) The minimum number of nodes in the cluster. Must be
```
Comment (kolea2) [inline google/cloud/bigtable/cluster.py:94]: I would reword this slightly to point out that this is a setting for autoscaling. Something like:
```
The miniumum number of nodes to be set in the cluster for autoscaling. Must be 1 or greater ...
```

File: google/cloud/bigtable/cluster.py [hunk 2]
```diff
@@ -52,7 +54,9 @@ class Cluster(object):
                         https://cloud.google.com/bigtable/docs/locations
 
     :type serve_nodes: int
-    :param serve_nodes: (Optional) The number of nodes in the cluster.
+    :param serve_nodes: (Optional) The number of nodes in the cluster. If any, of the
```
Comment (kolea2) [inline google/cloud/bigtable/cluster.py:57]: Slight rewording suggestion:
```
The number of nodes in the cluster for manual scaling.
```

File: google/cloud/bigtable/cluster.py [hunk 3]
```diff
@@ -104,10 +133,13 @@ def __init__(
         self.default_storage_type = default_storage_type
         self._kms_key_name = kms_key_name
         self._state = _state
+        self.min_serve_nodes = min_serve_nodes
```
Comment (kolea2) [inline google/cloud/bigtable/cluster.py:136]: do you think it would be a good idea to do some validation here? Here are some scenarios that come to mind:
- The user sets serve nodes AND autoscaling parameters
- The user sets no settings relating to auto or manual scaling
- The user only sets one or two (not all three) autoscaling config parameters

File: google/cloud/bigtable/cluster.py [hunk 4]
```diff
@@ -323,20 +366,61 @@ def update(self):
...
-                "location": None,
-            }
+
+        update_mask_pb = field_mask_pb2.FieldMask()
+
+        if self.serve_nodes:
+            update_mask_pb.paths.append("serve_nodes")
+
+        if self.min_serve_nodes or self.max_serve_nodes or self.cpu_utilization_percent:
+            update_mask_pb.paths.append("cluster_config")
```
Comment (kolea2) [inline google/cloud/bigtable/cluster.py:384]: I think we should add the more granular path - `cluster_config.cluster_autoscaling_config` and update `_to_pb()` to reflect that

File: tests/system/test_instance_admin.py [hunk 5]
```diff
@@ -614,3 +652,37 @@ def test_cluster_update(
     admin_cluster.serve_nodes = serve_nodes
     operation = admin_cluster.update()
     operation.result(timeout=60)  # Ensure the operation completes.
+
+
+def test_cluster_update_w_autoscaling(
```
Comment (kolea2) [inline tests/system/test_instance_admin.py:657]: looks good - I would also add a test going from auto -> manual scaling (disabling autoscaling)

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: tests/system/test_instance_admin.py (hunk 5 above)
Comment (Mariatta) [reply tests/system/test_instance_admin.py:657]: I added `test_cluster_disable_autoscaling` below.

File: google/cloud/bigtable/cluster.py (hunk 2 above)
Comment (Mariatta) [reply google/cloud/bigtable/cluster.py]: Updated.

File: google/cloud/bigtable/cluster.py (hunk 4 above)
Comment (Mariatta) [reply google/cloud/bigtable/cluster.py]: Updated.

File: google/cloud/bigtable/cluster.py (hunk 3 above)
Comment (Mariatta) [reply google/cloud/bigtable/cluster.py:135]: Great suggestion. I added the `_validate_scaling_config` function, which gets called in `create` and `update`. I also added unit tests to test the above scenarios.

What do you think about this?

File: google/cloud/bigtable/cluster.py [hunk 6]
```diff
@@ -211,6 +253,42 @@ def kms_key_name(self):
...
+        """Validate auto/manual scaling configuration before creating or updating."""
+
+        if (
+            not self.serve_nodes
+            and not self.min_serve_nodes
+            and not self.max_serve_nodes
+            and not self.cpu_utilization_percent
+        ):
+            raise ValueError(
+                "Must specify either serve_nodes or all of the autoscaling configurations."
```
Comment (kolea2) [inline google/cloud/bigtable/cluster.py:266]: maybe list the configurations here for convenience?

**Review by kolea2 — COMMENTED** [review]
lgtm after two small docs comments!

File: tests/system/test_instance_admin.py [hunk 7]
```diff
@@ -614,3 +652,69 @@ def test_cluster_update(
...
+
+    operation = admin_cluster_with_autoscaling.update()
+    operation.result(timeout=60)  # Ensure the operation completes.
+
+    # Create a new cluster instance and reload it.
+    alt_cluster = admin_instance_populated.cluster(admin_cluster_id)
+    alt_cluster.reload()
+    assert alt_cluster.min_serve_nodes == new_min_serve_nodes
+    assert alt_cluster.max_serve_nodes == new_max_serve_nodes
+    assert alt_cluster.cpu_utilization_percent == new_cpu_utilization_percent
```
Comment (kolea2) [inline tests/system/test_instance_admin.py:681]: let's add one more test case in which only one field is updated.

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: tests/system/test_instance_admin.py (hunk 7 above)
Comment (Mariatta) [reply tests/system/test_instance_admin.py:681]: I've added such test below (line 691)

**Review by kolea2 — APPROVED** [review]

**Review by kolea2 — APPROVED** [review]
